plugins/source/aws/codegen/recipes/base.go

```diff
@@ -418,3 +424,27 @@ func CreateReplaceTransformer(replace map[string]string) func(field reflect.Stru
 		return name, nil
 	}
 }
+
+func validateServiceMultiplex(multiplexerCall string) error {
+	re := regexp.MustCompile(`\"(.*?)\"`)
+	// Find the value of the service parameter
+	submatchAll := re.FindStringSubmatch(multiplexerCall)
+	t := client.ReadSupportedServiceRegions()
+
+	services := make(map[string]bool)
+
+	for _, partition := range t.Partitions {
+		for service := range partition.Services {
+			if _, ok := services[service]; !ok {
+				services[service] = true
+			}
+		}
+	}
+	if len(submatchAll) == 2 {
```

Comment:
For clarity why not run this check in line 432 (before the call to `ReadSupportedServiceRegions`) for a chance of early return? I don't think we actually have anything with no multiplexer set but it still makes sense: then here you don't need the `if` and it's all straightforward.

Reply:
Great call!... If a resource uses an Account multiplexer it doesn't have a service...